jfix-stdlib-concurrency/src/main/kotlin/ru/fix/stdlib/concurrency/events/EventReducer.kt

```diff
@@ -0,0 +1,76 @@
+package ru.fix.stdlib.concurrency.events
+
+import mu.KotlinLogging
+import java.util.concurrent.*
+
+
+/**
+ * EventReducer invokes given [handler] when [handle] function was invoked.
+ * If [handle] function was invoked (one or many times) during [handler] working,
+ * [handler] will be invoked after completing and so on.
+ *
+ * Therefore if [handle] function was invoked 100 times with breaks in 1 millisecond,
+ * and [handler]'s work takes 70 milliseconds,
+ * then [handler] wasn't invoked 100 times, but 3 times, consistently.
+ * */
+class EventReducer(
+        private val handler: () -> Unit,
+        private val shutdownCheckPeriodMs: Long = 1_000,
+        private val awaitTerminationPeriodMs: Long = 60_000
+) : AutoCloseable {
+    private val eventReceivingExecutor = Executors.newSingleThreadExecutor()
```

Comment:
What you think about using NamedExecutors?

Reply:
it helps to track threads state. Replaced  `Executors.newSingleThreadExecutor` with `NamedExecutors.newSingleThreadPool`